server/src/modules/users/users.service.ts

```diff
@@ -31,4 +33,16 @@ export class UsersService {
       throw error.code === '23505' ? new ConflictException('Email already exists') : new InternalServerErrorException();
     }
   }
+
+  async getRandomUsers(currentUserId: number): Promise<Array<UserResponse>> {
+    const users = await this.userRepository
+      .createQueryBuilder('user')
+      .where("user.id != :currentUserId", { currentUserId })
+      .orderBy("RANDOM()")
+      .limit(RANDOM_USERS_TO_GET)
+      .getMany()
+
+    const response = users.map(user => new UserResponse(user));
```

Comment:
No need to define a DTO just for this. You can reuse UserEntity class for the response. Just use plainToClass from class-transformer to exclude the password if needed.